### ISSUE TITLE
redis crash when series keys expire

### ISSUE THREAD

**Issue by Poluect** [RedisTimeSeries/RedisTimeSeries#528]
Hi! We are using rts with constantly having around 9k series keys. Each series key samples have 2 days retention period, and series itself has ttl 3 days (on each new sample ttl refreshed). Periodically (each 20s) data grapped from rts via ts.mrange. Exactly at a time ttl starts removing keys,  redis craches, most likely triggered somehow by `ts.mrange`, when after all keys loaded, but before labels fetched, some of these keys already removed by ttl

```
1:C 08 Sep 2020 01:19:41.547 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
1:C 08 Sep 2020 01:19:41.547 # Redis version=5.0.5, bits=64, commit=00000000, modified=0, pid=1, just started
1:C 08 Sep 2020 01:19:41.547 # Configuration loaded
1:M 08 Sep 2020 01:19:41.548 * Running mode=standalone, port=6379.
1:M 08 Sep 2020 01:19:41.548 # WARNING: The TCP backlog setting of 511 cannot be enforced because /proc/sys/net/core/somaxconn is set to the lower value of 128.
1:M 08 Sep 2020 01:19:41.548 # Server initialized
1:M 08 Sep 2020 01:19:41.548 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
1:M 08 Sep 2020 01:19:41.548 # WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with Redis. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. Redis must be restarted after THP is disabled.
1:M 08 Sep 2020 01:19:41.549 * Module 'timeseries' loaded from /usr/lib/redis/modules/redistimeseries.so
1:M 08 Sep 2020 01:19:41.549 * Reading RDB preamble from AOF file...
1:M 08 Sep 2020 01:19:44.924 * Reading the remaining AOF tail...
1:M 08 Sep 2020 01:20:23.819 * DB loaded from append only file: 42.270 seconds
1:M 08 Sep 2020 01:20:23.819 * Ready to accept connections
1:M 08 Sep 2020 01:20:32.894 # <timeseries> couldn't open key or key is not a Timeseries. key=temperature:2:32

...=== REDIS BUG REPORT START: Cut & paste starting from here ===
1:M 08 Sep 2020 01:20:32.894 # Redis 5.0.5 crashed by signal: 11
1:M 08 Sep 2020 01:20:32.894 # Crashed running the instruction at: 0x5633b06c3d49
1:M 08 Sep 2020 01:20:32.894 # Accessing address: 0x7ff487e3de98
1:M 08 Sep 2020 01:20:32.894 # Failed assertion: <no assertion failed> (<no file>:0)

------ STACK TRACE ------
EIP:
redis-server *:6379(raxIteratorNextStep+0x259)[0x5633b06c3d49]

Backtrace:
redis-server *:6379(logStackTrace+0x32)[0x5633b068d112]
redis-server *:6379(sigsegvHandler+0x9e)[0x5633b068d7ee]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x12730)[0x7ff48098b730]
redis-server *:6379(raxIteratorNextStep+0x259)[0x5633b06c3d49]
redis-server *:6379(raxNext+0x10)[0x5633b06c4cf0]
redis-server *:6379(RM_DictNextC+0x16)[0x5633b06b5fa6]
/usr/lib/redis/modules/redistimeseries.so(TSDB_mrange+0x241)[0x7ff4807a70b1]
redis-server *:6379(RedisModuleCommandDispatcher+0x56)[0x5633b06b96d6]
redis-server *:6379(call+0xa7)[0x5633b06492b7]
redis-server *:6379(processCommand+0x51e)[0x5633b0649b5e]
redis-server *:6379(processInputBuffer+0x171)[0x5633b0659d51]
redis-server *:6379(aeProcessEvents+0x101)[0x5633b06432e1]

Backtrace:
redis-server *:6379(logStackTrace+0x32)[0x5633b068d112]
redis-server *:6379(sigsegvHandler+0x9e)[0x5633b068d7ee]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x12730)[0x7ff48098b730]
redis-server *:6379(raxIteratorNextStep+0x259)[0x5633b06c3d49]
redis-server *:6379(raxNext+0x10)[0x5633b06c4cf0]
redis-server *:6379(RM_DictNextC+0x16)[0x5633b06b5fa6]
/usr/lib/redis/modules/redistimeseries.so(TSDB_mrange+0x241)[0x7ff4807a70b1]
redis-server *:6379(RedisModuleCommandDispatcher+0x56)[0x5633b06b96d6]
redis-server *:6379(call+0xa7)[0x5633b06492b7]
redis-server *:6379(processCommand+0x51e)[0x5633b0649b5e]
redis-server *:6379(processInputBuffer+0x171)[0x5633b0659d51]
redis-server *:6379(aeProcessEvents+0x101)[0x5633b06432e1]
redis-server *:6379(aeMain+0x2b)[0x5633b06436eb]
redis-server *:6379(main+0x4b9)[0x5633b0640569]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xeb)[0x7ff4807dc09b]
redis-server *:6379(_start+0x2a)[0x5633b06407aa]

------ INFO OUTPUT ------
# Server^M
redis_version:5.0.5^M
redis_git_sha1:00000000^M
redis_git_dirty:0^M
redis_build_id:5e8f7e5111134784^M
redis_mode:standalone^M
os:Linux 4.15.0-101-generic x86_64^M
arch_bits:64^M
multiplexing_api:epoll^M
atomicvar_api:atomic-builtin^M
gcc_version:8.3.0^M
```

**Comment by danni-m** [comment]
@Poluect thanks for the bug report, i tried to reproduce it without any luck (on redis 6).
Can you check if this happens on Redis 6? (there were a few fixes in the RAX implementation in the last year and I want to check if its related or not)

**Comment by Poluect** [comment]
Hi @danni-m, thanks for quick reply! Yeah I tried to reproduce the case locally with the same data on newer rts (used image 1.4.4), and it also crashed, but with OOM message, although the case I described didn't contain OOM error. In case of 1.4.4 version, at the time of crash 2k keys have been deleted by ttl (7k left). Not sure if these two cases are related though

```bash
1:C 08 Sep 2020 13:16:29.988 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
1:C 08 Sep 2020 13:16:29.988 # Redis version=6.0.5, bits=64, commit=00000000, modified=0, pid=1, just started
1:C 08 Sep 2020 13:16:29.988 # Configuration loaded
1:M 08 Sep 2020 13:16:29.990 * Running mode=standalone, port=6379.
1:M 08 Sep 2020 13:16:29.990 # WARNING: The TCP backlog setting of 511 cannot be enforced because /proc/sys/net/core/somaxconn is set to the lower value of 128.
1:M 08 Sep 2020 13:16:29.990 # Server initialized
1:M 08 Sep 2020 13:16:29.990 # WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with Redis. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. Redis must be restarted after THP is disabled.
1:M 08 Sep 2020 13:16:29.991 * <timeseries> RedisTimeSeries version 10404, git_sha=d3375e21e003c6603b056a1051b8efc91cb5a761
1:M 08 Sep 2020 13:16:29.991 * <timeseries> Redis version found by RedisTimeSeries : 6.0.5 - oss
1:M 08 Sep 2020 13:16:29.991 * Module 'timeseries' loaded from /usr/lib/redis/modules/redistimeseries.so
1:M 08 Sep 2020 13:16:29.994 * Reading RDB preamble from AOF file...
1:M 08 Sep 2020 13:16:29.994 * Loading RDB produced by version 5.0.5
1:M 08 Sep 2020 13:16:29.994 * RDB age 9730 seconds
1:M 08 Sep 2020 13:16:29.994 * RDB memory usage when created 131.53 Mb
1:M 08 Sep 2020 13:16:29.994 * RDB has an AOF tail
1:M 08 Sep 2020 13:16:33.950 * Reading the remaining AOF tail...
1:M 08 Sep 2020 13:16:37.573 * DB loaded from append only file: 7.582 seconds
1:M 08 Sep 2020 13:16:37.573 * Ready to accept connections
1:M 08 Sep 2020 13:16:55.190 # <timeseries> couldn't open key or key is not a Timeseries. key=temperature:2:32
1:M 08 Sep 2020 13:16:55.190 # <timeseries> couldn't open key or key is not a Timeseries. key=temperature:3:32
1:M 08 Sep 2020 13:16:55.190 # <timeseries> couldn't open key or key is not a Timeseries. key=temperature:4:32

1:M 08 Sep 2020 13:16:55.190 # <timeseries> couldn't open key or key is not a Timeseries. key=temperature:5:32

1:M 08 Sep 2020 13:16:55.190 # Out Of Memory allocating 18446744073709551550 bytes!


=== REDIS BUG REPORT START: Cut & paste starting from here ===
1:M 08 Sep 2020 13:16:55.190 # ------------------------------------------------
1:M 08 Sep 2020 13:16:55.190 # !!! Software Failure. Press left mouse button to continue
1:M 08 Sep 2020 13:16:55.190 # Guru Meditation: Redis aborting for OUT OF MEMORY #server.c:4876
1:M 08 Sep 2020 13:16:55.190 # (forcing SIGSEGV in order to print the stack trace)
1:M 08 Sep 2020 13:16:55.190 # ------------------------------------------------
1:M 08 Sep 2020 13:16:55.191 # Redis 6.0.5 crashed by signal: 11
1:M 08 Sep 2020 13:16:55.191 # Crashed running the instruction at: 0x561fef113874
1:M 08 Sep 2020 13:16:55.191 # Accessing address: 0xffffffffffffffff
1:M 08 Sep 2020 13:16:55.191 # Failed assertion: <no assertion failed> (<no file>:0)

------ STACK TRACE ------
EIP:
redis-server *:6379(_serverPanic+0x124)[0x561fef113874]

Backtrace:
redis-server *:6379(logStackTrace+0x32)[0x561fef115e02]
redis-server *:6379(sigsegvHandler+0x9e)[0x561fef1164de]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x12730)[0x7f1ec903c730]
redis-server *:6379(_serverPanic+0x124)[0x561fef113874]
redis-server *:6379(redisOutOfMemoryHandler+0x34)[0x561fef0cf814]
redis-server *:6379(zrealloc+0x69)[0x561fef0d9d99]
redis-server *:6379(raxIteratorAddChars+0x41)[0x561fef152211]
redis-server *:6379(raxIteratorNextStep+0x224)[0x561fef1524d4]
redis-server *:6379(raxNext+0x10)[0x561fef1534b0]
redis-server *:6379(RM_DictNextC+0x16)[0x561fef140f26]
/usr/lib/redis/modules/redistimeseries.so(TSDB_generic_mrange+0x295)[0x7f1ec8e562d5]
redis-server *:6379(RedisModuleCommandDispatcher+0x54)[0x561fef146114]
redis-server *:6379(call+0x9d)[0x561fef0d1ffd]
redis-server *:6379(processCommand+0x33f)[0x561fef0d278f]
redis-server *:6379(processCommandAndResetClient+0x10)[0x561fef0e0480]
redis-server *:6379(processInputBuffer+0x18f)[0x561fef0e4acf]
redis-server *:6379(+0xd5fac)[0x561fef161fac]
redis-server *:6379(aeProcessEvents+0x2e7)[0x561fef0cbbd7]
redis-server *:6379(aeMain+0x1d)[0x561fef0cbf1d]
redis-server *:6379(main+0x4c9)[0x561fef0c87c9]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xeb)[0x7f1ec8e8b09b]
redis-server *:6379(_start+0x2a)[0x561fef0c8a5a]

------ INFO OUTPUT ------
# Server
redis_version:6.0.5
redis_git_sha1:00000000
redis_git_dirty:0
redis_build_id:a552bef8bac6cf1c
redis_mode:standalone
os:Linux 4.19.76-linuxkit x86_64
arch_bits:64
multiplexing_api:epoll
atomicvar_api:atomic-builtin
gcc_version:8.3.0
process_id:1
run_id:bfe847c436aa556aa7f563443fcd9587f21c8d24
tcp_port:6379
uptime_in_seconds:26
uptime_in_days:0
hz:10
configured_hz:10
lru_clock:5735495
executable:/data/redis-server
config_file:

# Clients
connected_clients:2
client_recent_max_input_buffer:2
client_recent_max_output_buffer:6614064
blocked_clients:0
tracking_clients:0
clients_in_timeout_table:0

# Memory
used_memory:114069256
used_memory_human:108.78M
used_memory_rss:153137152
used_memory_rss_human:146.04M
used_memory_peak:144956056
used_memory_peak_human:138.24M
used_memory_peak_perc:78.69%
used_memory_overhead:1591656
used_memory_startup:806800
used_memory_dataset:112477600
used_memory_dataset_perc:99.31%
allocator_allocated:119159640
allocator_active:124444672
allocator_resident:149626880
total_system_memory:2087469056
total_system_memory_human:1.94G
used_memory_lua:37888
used_memory_lua_human:37.00K
used_memory_scripts:0
used_memory_scripts_human:0B
number_of_cached_scripts:0
maxmemory:0
maxmemory_human:0B
maxmemory_policy:noeviction
allocator_frag_ratio:1.04
allocator_frag_bytes:5285032
allocator_rss_ratio:1.20
allocator_rss_bytes:25182208
rss_overhead_ratio:1.02
rss_overhead_bytes:3510272
mem_fragmentation_ratio:1.29
mem_fragmentation_bytes:34172816
mem_not_counted_for_evict:572
mem_replication_backlog:0
mem_clients_slaves:0
mem_clients_normal:33972
mem_aof_buffer:572
mem_allocator:jemalloc-5.1.0
active_defrag_running:0
lazyfree_pending_objects:0

# Persistence
loading:0
rdb_changes_since_last_save:691366
rdb_bgsave_in_progress:0
rdb_last_save_time:1599570989
rdb_last_bgsave_status:ok
rdb_last_bgsave_time_sec:-1
rdb_current_bgsave_time_sec:-1
rdb_last_cow_size:0
aof_enabled:1
aof_rewrite_in_progress:0
aof_rewrite_scheduled:0
aof_last_rewrite_time_sec:-1
aof_current_rewrite_time_sec:-1
aof_last_bgrewrite_status:ok
aof_last_write_status:ok
aof_last_cow_size:0
module_fork_in_progress:0
module_fork_last_cow_size:0
aof_current_size:398050859
aof_base_size:396277836
aof_pending_rewrite:0
aof_buffer_length:570
aof_rewrite_buffer_length:0
aof_pending_bio_fsync:0
aof_delayed_fsync:0

# Stats
total_connections_received:12
total_commands_processed:9389
instantaneous_ops_per_sec:0
total_net_input_bytes:1409849
total_net_output_bytes:17126151
instantaneous_input_kbps:0.00
instantaneous_output_kbps:0.00
rejected_connections:0
sync_full:0
sync_partial_ok:0
sync_partial_err:0
expired_keys:1659
expired_stale_perc:15.03
expired_time_cap_reached_count:0
expire_cycle_cpu_milliseconds:95
evicted_keys:0
keyspace_hits:27878
keyspace_misses:5
pubsub_channels:0
pubsub_patterns:0
latest_fork_usec:0
migrate_cached_sockets:0
slave_expires_tracked_keys:0
active_defrag_hits:0
active_defrag_misses:0
active_defrag_key_hits:0
active_defrag_key_misses:0
tracking_total_keys:0
tracking_total_items:0
tracking_total_prefixes:0
unexpected_error_replies:0

# Replication
role:master
connected_slaves:0
master_replid:ac4745d10bc6394084bc92553cb2b0bb403c459e
master_replid2:0000000000000000000000000000000000000000
master_repl_offset:0
second_repl_offset:-1
repl_backlog_active:0
repl_backlog_size:1048576
repl_backlog_first_byte_offset:0
repl_backlog_histlen:0

# CPU
used_cpu_sys:3.245441
used_cpu_user:7.015924
used_cpu_sys_children:0.001100
used_cpu_user_children:0.003477

# Modules
module:name=timeseries,ver=10404,api=1,filters=0,usedby=[],using=[],options=[]

# Commandstats
cmdstat_expire:calls=9286,usec=376217,usec_per_call=40.51
cmdstat_ts.mrange:calls=3,usec=358922,usec_per_call=119640.66
cmdstat_scan:calls=93,usec=9583,usec_per_call=103.04
cmdstat_ping:calls=4,usec=36,usec_per_call=9.00
cmdstat_zremrangebyscore:calls=1,usec=6,usec_per_call=6.00
cmdstat_info:calls=1,usec=38,usec_per_call=38.00
cmdstat_zrangebyscore:calls=1,usec=10,usec_per_call=10.00

# Cluster
cluster_enabled:0

# Keyspace
db0:keys=7628,expires=7627,avg_ttl=207553

------ CLIENT LIST OUTPUT ------
id=6 addr=172.18.0.1:33050 fd=10 name= age=25 idle=5 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 obl=0 oll=0 omem=0 events=r cmd=zrangebyscore user=default
id=7 addr=172.18.0.1:33054 fd=11 name= age=22 idle=0 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=164 qbuf-free=32604 obl=0 oll=41 omem=14408 events=r cmd=ts.mrange user=default

------ CURRENT CLIENT INFO ------
id=7 addr=172.18.0.1:33054 fd=11 name= age=22 idle=0 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=164 qbuf-free=32604 obl=0 oll=41 omem=14408 events=r cmd=ts.mrange user=default
argv[0]: 'TS.MRANGE'
argv[1]: '1599570415180'
argv[2]: '1599571015180'
argv[3]: 'AGGREGATION'
argv[4]: 'AVG'
argv[5]: '60000'
argv[6]: 'COUNT'
argv[7]: '100000'
argv[8]: 'WITHLABELS'
argv[9]: 'FILTER'
argv[10]: 'type=local'

------ REGISTERS ------
1:M 08 Sep 2020 13:16:55.193 # 
RAX:0000000000000000 RBX:00007fff07addf90
RCX:00007f1ec8f51537 RDX:0000000000000000
RDI:0000000000000000 RSI:00007f1ec90248c0
RBP:0000561fef1f44ce RSP:00007fff07addf70
R8 :0000000000000000 R9 :00007f1ec8e64f80
R10:0000000000000000 R11:0000000000000000
R12:000000000000130c R13:0000000000000001
R14:00007f1ec533e0e4 R15:00007f1ec8a30438
RIP:0000561fef113874 EFL:0000000000010202
CSGSFS:002b000000000033
1:M 08 Sep 2020 13:16:55.193 # (00007fff07addf7f) -> 00007f1ec8e626e0
1:M 08 Sep 2020 13:16:55.193 # (00007fff07addf7e) -> 0000000000000038
1:M 08 Sep 2020 13:16:55.193 # (00007fff07addf7d) -> 00007f1ec57d35b0
1:M 08 Sep 2020 13:16:55.193 # (00007fff07addf7c) -> 0000000000000002
1:M 08 Sep 2020 13:16:55.193 # (00007fff07addf7b) -> 0000000000000001
1:M 08 Sep 2020 13:16:55.193 # (00007fff07addf7a) -> 0000000000000001
1:M 08 Sep 2020 13:16:55.193 # (00007fff07addf79) -> 000000007bdf0000
1:M 08 Sep 2020 13:16:55.193 # (00007fff07addf78) -> 0000000000000000
1:M 08 Sep 2020 13:16:55.193 # (00007fff07addf77) -> 59524f4d454d2046
1:M 08 Sep 2020 13:16:55.193 # (00007fff07addf76) -> 4f2054554f20726f
1:M 08 Sep 2020 13:16:55.193 # (00007fff07addf75) -> 6620676e6974726f
1:M 08 Sep 2020 13:16:55.193 # (00007fff07addf74) -> 6261207369646552
1:M 08 Sep 2020 13:16:55.193 # (00007fff07addf73) -> 00007fff07ade090
1:M 08 Sep 2020 13:16:55.194 # (00007fff07addf72) -> 00007fff07ade160
1:M 08 Sep 2020 13:16:55.194 # (00007fff07addf71) -> 0000003000000018
1:M 08 Sep 2020 13:16:55.194 # (00007fff07addf70) -> 0000003000000020

------ MODULES INFO OUTPUT ------

------ FAST MEMORY TEST ------
1:M 08 Sep 2020 13:16:55.194 # Bio thread for job type #0 terminated
1:M 08 Sep 2020 13:16:55.194 # Bio thread for job type #1 terminated
1:M 08 Sep 2020 13:16:55.195 # Bio thread for job type #2 terminated
*** Preparing to test memory region 561fef261000 (2277376 bytes)
*** Preparing to test memory region 561fefa35000 (1351680 bytes)
*** Preparing to test memory region 7f1eb4000000 (135168 bytes)
*** Preparing to test memory region 7f1ebb200000 (8388608 bytes)
*** Preparing to test memory region 7f1ebba00000 (62914560 bytes)
*** Preparing to test memory region 7f1ebf6fc000 (116391936 bytes)
*** Preparing to test memory region 7f1ec65fd000 (8388608 bytes)
*** Preparing to test memory region 7f1ec6dfe000 (8388608 bytes)
*** Preparing to test memory region 7f1ec75ff000 (8388608 bytes)
*** Preparing to test memory region 7f1ec7e00000 (8388608 bytes)
*** Preparing to test memory region 7f1ec8600000 (8388608 bytes)
*** Preparing to test memory region 7f1ec8e63000 (16384 bytes)
*** Preparing to test memory region 7f1ec9024000 (24576 bytes)
*** Preparing to test memory region 7f1ec9047000 (16384 bytes)
*** Preparing to test memory region 7f1ec9330000 (16384 bytes)
*** Preparing to test memory region 7f1ec9558000 (8192 bytes)
*** Preparing to test memory region 7f1ec9585000 (4096 bytes)
.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O.O
Fast memory test PASSED, however your memory can still be broken. Please run a memory test for several hours if possible.

------ DUMPING CODE AROUND EIP ------
Symbol: _serverPanic (base: 0x561fef113750)
Module: redis-server *:6379 (base 0x561fef08c000)
$ xxd -r -p /tmp/dump.hex /tmp/dump.bin
$ objdump --adjust-vma=0x561fef113750 -D -b binary -m i386:x86-64 /tmp/dump.bin
------
1:M 08 Sep 2020 13:16:55.742 # dump of function (hexdump of 420 bytes):
41544189f4554889fd534881ecd001000048898c24380100004c898424400100004c898c244801000084c074400f298424500100000f298c24600100000f299424700100000f299c24800100000f29a424900100000f29ac24a00100000f29b424b00100000f29bc24c0010000488d5c2420488d4c2408be00010000c744240818000000488d8424f00100004889dfc744240c300000004889442410488d8424200100004889442418e8623dfbff8b053047150085c00f8484000000488d35cd080f00bf0300000031c0e8e1befbff488d35f2080f00bf0300000031c0e8cebefbff4589e04889e94889da488d3571210f00bf0300000031c0e8b2befbff488d3503090f00bf0300000031c0e89fbefbff488d3578080f00bf0300000031c0e88cbefbffc60425ffffffff784881c4d00100005b5d415cc30f1f840000000000e8fbf4ffffe972ffffff660f1f4400004157415641554989cd41544989f4554889fd534889d3ba040000004889df4881ec080300000fb601488d74243483e00f0fc889442434e8f5f5ffff4c89e64889efe8ba7afdff4889442408410fb6450089c283e2
Function at 0x561fef0cf700 is serverLog
Function at 0x561fef112ed0 is mixDigest
Function at 0x561fef0eb3a0 is getExpire

=== REDIS BUG REPORT END. Make sure to include from START to END. ===

       Please report the crash by opening an issue on github:

           http://github.com/antirez/redis/issues

  Suspect RAM error? Use redis-server --test-memory to verify it.
```

**Comment by gkorland** [comment]
@Poluect did the issue remain?

**Comment by Poluect** [comment]
yeah, still happens. At first I thought OOM issue I described above was just a local case, as there is a limited memory for docker, but memory spike was actually huge
```bash
1:M 08 Sep 2020 13:16:55.190 # Out Of Memory allocating 18446744073709551550 bytes!
```
18446744 terabytes, and the case also reproduced on redis 6.0.5

**Comment by jeffreylovitz** [comment]
@Poluect This issue should now be resolved on the master branch due to the fix introduced in #549.

**Comment by Poluect** [comment]
@jeffreylovitz awesome, thanks!

**Comment by danni-m** [comment]
@Poluect please reopen if you continue to experience the issue after the fix was applied (we are going to release 1.4.6 that will contain this fix)
